apps/openmw/mwmechanics/recharge.cpp

```diff
@@ -0,0 +1,92 @@
+#include "recharge.hpp"
+
+#include <components/misc/rng.hpp>
+
+#include "../mwbase/world.hpp"
+#include "../mwbase/environment.hpp"
+#include "../mwbase/windowmanager.hpp"
+
+#include "../mwworld/containerstore.hpp"
+#include "../mwworld/class.hpp"
+#include "../mwworld/esmstore.hpp"
+
+#include "creaturestats.hpp"
+#include "actorutil.hpp"
+
+namespace MWMechanics
+{
+
+bool rechargeItem(const MWWorld::Ptr &item, const float maxCharge, const float duration)
+{
+    float charge = item.getCellRef().getEnchantmentCharge();
+    if (charge == -1 || charge == maxCharge)
+        return false;
+
+    static float fMagicItemRechargePerSecond = MWBase::Environment::get().getWorld()->getStore().get<ESM::GameSetting>().find(
```

Comment:
This should either be "const float" or "static const float" because if it's static but not const the overhead of the record recovery will remain with the GMST remaining in memory for no reason and if we care about making GMSTs not read-only in future leaving it "const float" is enough.